lib/connection_string.js

```diff
@@ -580,6 +613,12 @@ function parseConnectionString(uri, options, callback) {
     return callback(parseError);
   }
 
+  try {
+    checkTLSOptions(parsedOptions);
```

Comment:
Is there a reason you didn't make this call within `parseQueryString`? Seems like it would prevent an extra `try/catch` block

Reply:
@mbroadst I don't believe I can do that because the that function doesn't have access to the complete `options` object.

I actually meant to put the `checkTLSOptions` a couple lines lower below `parsedOptions = Object.assign({}, parsedOptions, options);` The idea being there was a check specifically for `queryOptions` and then one for `options`. 

I think this piece of code is a bit confusing with the reassignments. I see it as three pieces (not named this way) `queryOptions`, `passedOptions` and `options` (both "merged").

Reply:
Hmm, maybe I'm confused. It's my understanding that we are only trying to validate that conflicting query string parameters are not specified, similarly with options passed into the `MongoClient` constructor. We shouldn't need to depend on any merged options at this point, `parseQueryString` takes in the `query` and `options` so I would have assumed that was all the detail we need to complete the task here.